frame/support/src/traits.rs

```diff
@@ -867,6 +867,12 @@ pub trait LockableCurrency<AccountId>: Currency<AccountId> {
 	/// The quantity used to denote time; usually just a `BlockNumber`.
 	type Moment;
 
+	/// Retrieve lock information for `id`, returning `None` if no lock exists.
+	fn get_lock(
```

Comment:
this should probably take `(who: &AccountId, reasons: WithdrawReasons, maybe_exclude: Option<LockIdentifier>) -> Option<Self::Balance>`

Reply:
perhaps `reasons: Option<WithdrawReasons>`?